tmva/sofie/src/RModel.cxx

```diff
@@ -164,15 +170,36 @@ namespace SOFIE{
       }
    }
 
-   void RModel::Initialize(){
+   void RModel::Initialize(int batchSize){
+      // check if there are only parametrized input tensor and convert in
+      // ready input tensor according to batch size
+      // convert parametric shape to a dimensional shape
+      if (fReadyInputTensorInfos.size() != fInputTensorNames.size()) {
+         if ( fReadyInputTensorInfos.size() + fInputTensorInfos.size() != fInputTensorNames.size())
+            throw std::runtime_error("TMVA-SOFIE: RModel::Initializes: invalid inputs");
+         for (auto & input : fInputTensorInfos) {
+            std::vector<size_t> shape;
+            shape.reserve(input.second.shape.size());
+            for (auto & d : input.second.shape){
+               if (d.isParam)
+                  shape.push_back(batchSize);
```

Comment:
While for most of the use cases, if there is an input shape parameter it is going to be the batch size, in ONNX standard there can be other possibilities. for e.g. a CNN model with variable input image size. Should we hard-code this behaviour here, or is this just a temporary measure?

Reply:
Good point !
Is there a way to know from ONNX if the parametrised input is the batch size or something else ?